src/workloads/c2c/MongosyncScripts.yml

```diff
@@ -12,88 +12,35 @@ Description: |
         Path: ./MongosyncScripts.yml
         Key: StartMongosync
 
-  Note that the polling scripts start with set -eou pipefail so that any
-  subcommand run during the script failing will cause the entire script to fail.
-  This way if mongosync crashes during the test, it will stop the test execution
-  immediately and genny will bubble up the error and fail the test
-
-  For curl commands we pass -s and -S so that an errors are still reported (-S)
-  but download progress stats are not logged to stderr
-
 StartMongosync:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: StartMongosync
-  Script: |
-    curl -s -S -X POST http://10.2.0.160:27182/api/v1/start \
-      -H 'Content-Type: application/json' \
-      -d '{"Source": "cluster0", "Destination": "cluster1"}'
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m start
```

Comment:
Are these patches going to be brittle? It looks like this will only run if we are in WORKDIR, which assumes src/genny within it, but it won't work from within src/genny if you ran Genny direction with ./run-genny. 

Why not be match the behavior of LoadConfig and load relative to this file?

Reply:
If we are already passing in `-m start`, why not explicitly pass the connection url through the command line also, rather than the environmental variable? Or write out and read in a json file?

Reply:
Had an idea - what if we put the python script in a top-level folder like `cast_python` beside `cast_core`, and add `cast_python` to the python path in `run-genny`?

Then we can do something like this:

```
StartMongosync:
  Repeat: 1
  Command: python3
  MetricsName: StartMongosync
  Script: |
    import mongosync_actor as actor
    actor.change_state("/api/v1/start",
        {"Source": "cluster0", "Destination": "cluster1"})
```

Reply:
I took a stab at the `cast_python` idea and uploaded a new change list. Let me know what you think.

I also set it up so we can write unit tests for the python actors and so that they get linted.

Reply:
FYI @rtimmons I asked for script paths to be relative to workload rather than work directory in this thread. The work directory is a DSI concept, and it seems bad if Genny can't be run locally outside of DSI.

Reply:
Strong disagree - paths are notoriously confusing everywhere throughout our ecosystem. Plus having them relative to the workload makes it harder to have them cross-repo. Having all paths everywhere be strictly relative to working directory makes everything simple and obvious.

The "work dir" concept is really just a workspace concept. A convention for how to check out repos. I don't think it's necessarily a DSI concept. See related jira ticket [here](https://jira.mongodb.org/browse/EVG-18189).

(Strong opinion weakly held here - happy to have a further convo, but I think this is a super important concept to get right.)

Reply:
I definitely agree with `paths are notoriously confusing everywhere throughout our ecosystem.`. At the same time I would find it surprising if I couldn't checkout out genny and 
```
cd genny
./run-genny install
./run-genny workload myworkload
```
I think that would mess with people's built in assumptions.

Reply:
> I think that would mess with people's built in assumptions.

That's what we gotta change 🥹 Multi-repo setups are a nightmare unless we have a consistent way to manage them and for repos to know where each other are. I think `run-genny` and `run-dsi` could guide users to this, and it becomes pretty natural. This is, after all, how DSI and Genny are run from CI, and local mirroring CI is goodness.

Reply:
Another angle that might be more convincing: Imagine wanting the genny yaml and the script to live in separate repos. You'd need both repos to be checked out to run locally. How would you coordinate those checkouts otherwise? You could check it out inside the genny repo but then do you still use evergreen modules?

Reply:
Some more thoughts:

1. These scripts are not in separate repos. They are in the same repo. If we use relative paths for in the same repo (current problem), it doesn't matter where you call ./run-genny at all. This would also be consistent to how we handle load_config. 
2. I grant that cross repo scripts would be an entirely different ball of wax, but that's not what we have here. 
3. The original purpose of the work directory was to free the user from having to run in a pre-ordained place. We seem to be moving back to having to work in a pre-ordained location or else things will fail in non-helpful ways. 

I have some other review notes to send also -- let me finish my open review, and then we can talk about how we get Brad and this work over the line.

Reply:
For the Genny case here, do any of us imagine a use case in which cross repo scripts exists and are called from within a workload? 

If we do, I'd suggest we consider supporting relative and absolute paths with different names. If we can't, I say we focus on the one repo case.